client/src/App.tsx

```diff
@@ -84,83 +106,105 @@ const App: React.FC = () => {
+              <input type="password" name="password" placeholder="Password" onChange={handleInputChange} />
+              <input type="text" name="database" placeholder="Database Name" onChange={handleInputChange} />
               <div>
-                <h2>Table Data:</h2>
-                <table>
-                  <thead>
-                    <tr>
-                      <th>ID</th>
-                      <th>Name</th>
-                    </tr>
-                  </thead>
-                  <tbody>
-                    {selectedTable.filter((_, i) => i >= PAGE_LIMIT * (currentPage - 1) && i < PAGE_LIMIT * currentPage).map(row => (
-                      <tr key={row.id}>
-                        <td>{row.id}</td>
-                        <td>{row.name}</td>
-                      </tr>
-                    ))}
-                  </tbody>
-                </table>
-                <CRow>
-                  <CCol xs={6} md={6} xl={6}/>
-                  <CCol xs={6} md={6} xl={6}>
-                      <CSmartPagination
-                          align="end"
-                          activePage={currentPage}
-                          pages={totalPage}
-                          onActivePageChange={setCurrentPage}
-                          size="sm"
-                          limit={2}
-                      />
-                  </CCol>
-                </CRow>
-                {/* <div>
-                  <ul className="pagination">
-                  {Array.from({ length: Math.ceil(selectedTable.length / itemsPerPage) }, (_, index) => index).map(number => (
-                      <li key={number} className="page-item">
-                        <button onClick={() => paginate(number + 1)} className="page-link">
-                          {number + 1}
-                        </button>
-                      </li>
-                    ))}
-                  </ul>
-                </div> */}
+                <button onClick={handleTestConnection}>Test Connection</button>
+                <button onClick={handleConnect}>Connect</button>
               </div>
-            }
+              <textarea value={sqlQuery} onChange={(e) => setSqlQuery(e.target.value)} />
+              <button onClick={handleRunQuery}>Run Query</button>
+              <Suspense fallback={<LoadingBar />}>
+                {error && <div>{error}</div>}
+                {queryResult.length > 0 && (
+                  <div>
+                    <h2>Query Result:</h2>
+                    <table>
+                      <thead>
+                        <tr>
+                          {Object.keys(queryResult[0]).map((key) => (
+                            <th key={key}>{key}</th>
+                          ))}
+                        </tr>
+                      </thead>
+                      <tbody>
+                        {queryResult.map((row, rowIndex) => (
+                          <tr key={rowIndex}>
```

Comment:
While rendering array of data in React, it's not a good way to use index as key.
You need to use some kind of unique identifier instead.